api/src/main/scala/JobUtils.scala

```diff
@@ -0,0 +1,140 @@
+  val baseLayerReader = S3CollectionLayerReader("datahub-catalogs-us-east-1", "")
+
+  /**
+    * For an InputJson object, return a histogram of raster grouped count results.
+    *
+    * @param   crs    The key (e.g. "input.rasterCRS")
+    * @param   input  The InputJson
+    * @return         A histogram of results in JSON format
+    */
+  def getRasterGroupedCount(input: InputData) = {
+    val rasterCRS = getCRS(input.rasterCRS)
+    val polygonCRS = getCRS(input.polygonCRS)
+    val rasterLayerIds = input.rasters.map { str => LayerId(str, input.zoom) }
+    val polygons = input.polygon.map { str =>
+      parseGeometry(str, polygonCRS, rasterCRS).buffer(0).asMultiPolygon.get
+    }
+    val aoi = polygons.unionGeometries.asMultiPolygon.get
+    val rasterLayers = rasterLayerIds.map { layer => fetchLayer(layer, aoi) }
+    val count =
+      rasterGroupedCount(rasterLayers, aoi)
+        .mapValues(_.sum().toInt)
+        .map { case (k, v) => k.toString -> v}
+        .toMap
+
+    Map("result" -> count).toJson
+  }
+
+  /**
+    * Transform the incoming GeoJSON into a [[MultiPolygon]] in the
+    * destination CRS.
+    *
+    * @param   geoJson  The incoming geometry
+    * @param   srcCRS   The CRS that the incoming geometry is in
+    * @param   destCRS  The CRS that the outgoing geometry should be in
+    * @return           A MultiPolygon
+    */
+  def parseGeometry(geoJson: String, srcCRS: CRS, destCRS: CRS): MultiPolygon = {
+    geoJson.parseJson.convertTo[Geometry] match {
+      case p: Polygon => MultiPolygon(p.reproject(srcCRS, destCRS))
+      case mp: MultiPolygon => mp.reproject(srcCRS, destCRS)
+      case _ => MultiPolygon()
+    }
+  }
+
+  /**
+    * For a given config and CRS key, return one of several recognized
+    * [[geotrellis.proj4.CRS]]s, or raise an error.
+    *
+    * @param   crs  The key (e.g. "input.rasterCRS")
+    * @return       A CRS
+    */
+  private def getCRS(crs: String): CRS = crs match {
+    case "LatLng" => LatLng
+    case "WebMercator" => WebMercator
+    case "ConusAlbers" => ConusAlbers
+    case s: String => throw new Exception(s"Unknown CRS: $s")
+  }
+
+  /**
+    * For a layerId and a shape, retrieve an intersecting TileLayerCollection.
+    *
+    * @param   layerId  The LayerId
+    * @param   shape    The shape as a MultiPolygon
+    * @return           A TileLayerCollection intersecting the shape
+    */
+  def fetchLayer(layerId: LayerId, shape: MultiPolygon): TileLayerCollection[SpatialKey] =
+    baseLayerReader
+      .query[SpatialKey, Tile, TileLayerMetadata[SpatialKey]](layerId)
+      .where(Intersects(shape))
+      .result
+
+  /**
+    * From a sequence of layers, return a Map of Tile sequences.
+    *
+    * @param   layers  A sequence of TileLayerCollections
+    * @return          A map of Tile sequences, keyed with the SpatialKey
+    */
+  type TilesMap = Map[SpatialKey, Seq[Tile]]
+  type TileCollectionSeq = Seq[TileLayerCollection[SpatialKey]]
```

Comment:
I don't think these type names make the code simpler. In addition to the proper types in the API, the reader now has to keep in mind these aliases. We can use formatting to make the code more readable while keeping the original types:

```scala
def joinCollectionLayers(
  layers: Seq[TileLayerCollection[SpatialKey]]
): Map[SpatialKey, Seq[Tile]] = {
  
  val maps: Seq[Map[SpatialKey, Tile]] = layers.map((_: Seq[(SpatialKey, Tile)]).toMap)
  val keySet: Array[SpatialKey] = maps.map(_.keySet).reduce(_ union _).toArray
  for (key: SpatialKey <- keySet) yield {
    val tiles: Seq[Tile] = maps.map(_.apply(key))
    key -> tiles
  }
}.toMap
```